scripts/partition_manager.py

```diff
@@ -24,12 +24,42 @@ def item_is_placed(d, item, after_or_before):
            d['placement'][after_or_before][0] == item
 
 
+def resolve_one_of(reqs, partitions):
+    for k, v in reqs.items():
+        if isinstance(v, dict):
+            if 'one_of' in v.keys():
+                if len(v.keys()) != 1:
+                    raise RuntimeError("'one_of' must be the only key in its dict")
+                # Now fetch the first existing partition. Note that the value must be a list even if there is only
+                # one entry.
+                reqs[k] = [partition for partition in v['one_of'] if partition in partitions][:1]
+            else:
+                resolve_one_of(v, partitions)
+        # 'one_of' dicts can occur inside lists of partitions.
+        # dicts with 'one_of' key is the only dict supported inside lists
+        elif isinstance(v, list):
+            # Resolve all 'one-of' dicts inside the list
+            to_remove = list()
+            to_add = list()
+            for i in v:
+                if isinstance(i, dict):
+                    if 'one_of' not in i.keys():
+                        raise RuntimeError("Found illegal dict inside list. Only 'one_of' dicts are allowed")
+                    to_add.append([partition for partition in i['one_of'] if partition in partitions][0])
```

Comment:
If there are no existing entries in the 'one_of' list, this fails with an IndexError or something. Can it fail more gracefully?

Reply:
Will now print specifically why it failed through a runtimeerror